BDArmory/Weapons/ModuleWeapon.cs

```diff
@@ -3271,12 +3149,14 @@ bool CanFire(float AmmoPerShot)
                     //else return true; //this is causing weapons thath have ECPerShot + standard ammo (railguns, etc) to not consume ammo, only EC
```

Comment:
The SecondaryAmmoPerShot section above should also get the ``BDArmorySettings.WEAPONS_RESPECT_CROSSFEED`` changes, on the off chance someone makes a custom gun that uses something other than ElectricCharge as a secondary resource.